src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go

```diff
@@ -104,3 +112,27 @@ func (rp *RequestPool) currentSyncForUser(req syncRequest, currentPos types.Stre
 	}
 	return rp.db.IncrementalSync(req.userID, req.since, currentPos, req.limit)
 }
+
+func (rp *RequestPool) appendAccountData(data *types.Response, userID string) (*types.Response, error) {
+	localpart, _, err := gomatrixserverlib.SplitID('@', userID)
+	if err != nil {
+		return nil, err
+	}
+
+	events, err := rp.accountDB.GetAccountData(localpart, "")
+	if err != nil {
+		return nil, err
+	}
+	data.AccountData.Events = events
+
+	for r, j := range data.Rooms.Join {
+		events, err := rp.accountDB.GetAccountData(localpart, r)
```

Comment:
It might be nice to do this as a single bulk lookup that returns all the account data for a single user. That way we would have fewer round trips to the database.